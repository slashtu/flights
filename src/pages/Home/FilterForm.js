import React from 'react';
import PropTypes from 'prop-types';
import { Field, reduxForm } from 'redux-form';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import MenuItem from '@material-ui/core/MenuItem';

import {
  TYPES,
  ID_LABEL,
  ARRIVAL_LABEL,
  DEPARTURE_LABEL,
  DATE_LABEL,
  TYPE_LABEL,
  TYPE_ALL
} from 'constants/flight';

import styles from './FilterForm.css';

const types = [...TYPES, TYPE_ALL].map(t => ({
  label: t.toUpperCase(),
  value: t
}));

const renderTextField = ({ input, label }) => (
  <TextField label={label} fullWidth {...input} />
);

const renderSelectField = ({ input, label, items }) => (
  <TextField select label={label} {...input} fullWidth>
    {items.map(option => (
      <MenuItem key={option.value} value={option.value}>
        {option.label}
      </MenuItem>
    ))}
  </TextField>
);

const renderDatePicker = ({ input, label }) => (
  <TextField
    label={label}
    type="date"
    InputLabelProps={{
      shrink: true
    }}
    fullWidth
    {...input}
  />
);

const FilterForm = props => {
  const { handleSubmit } = props;
  return (
    <form className={styles.container} onSubmit={handleSubmit}>
      <Grid container spacing={24}>
        <Grid item xs={6} sm={2}>
          <Field name="id" component={renderTextField} label={ID_LABEL} />
        </Grid>
        <Grid item xs={6} sm={2}>
          <Field
            name="departure"
            component={renderTextField}
            label={DEPARTURE_LABEL}
          />
        </Grid>
        <Grid item xs={6} sm={2}>
          <Field
            name="arrival"
            component={renderTextField}
            label={ARRIVAL_LABEL}
          />
        </Grid>
        <Grid item xs={6} sm={2}>
          <Field
            name="type"
            items={types}
            component={renderSelectField}
            label={TYPE_LABEL}
          />
        </Grid>
        <Grid item xs={6} sm={2}>
          <Field name="date" label={DATE_LABEL} component={renderDatePicker} />
        </Grid>
      </Grid>
    </form>
  );
};

FilterForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

export default reduxForm({ form: 'filter' })(FilterForm);
