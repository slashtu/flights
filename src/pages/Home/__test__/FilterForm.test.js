import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import FilterForm from '../FilterForm';

const mockStore = configureMockStore();

describe('<FilterForm />', () => {
  it('Component snapshot [shallow]', () => {
    const store = mockStore({});
    const wrapper = shallow(
      <Provider store={store}>
        <BrowserRouter>
          <FilterForm />
        </BrowserRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
