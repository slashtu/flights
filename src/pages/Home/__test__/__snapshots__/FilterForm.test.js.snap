// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterForm /> Component snapshot [shallow] 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <BrowserRouter>
    <ReduxForm
      destroyOnUnmount={true}
      enableReinitialize={false}
      forceUnregisterOnUnmount={false}
      form="filter"
      getFormState={[Function]}
      keepDirtyOnReinitialize={false}
      persistentSubmitErrors={false}
      pure={true}
      shouldAsyncValidate={[Function]}
      shouldError={[Function]}
      shouldValidate={[Function]}
      shouldWarn={[Function]}
      touchOnBlur={true}
      touchOnChange={false}
      updateUnregisteredFields={false}
    />
  </BrowserRouter>
</ContextProvider>
`;
